src/test/java/com/github/krzysiekjodlowski/ox/BoardValidatorHorizontalCheckTest.java

```diff
@@ -0,0 +1,219 @@
+        //arrange
+        Board board = new Board(4);
+        BoardValidator boardValidator = new BoardValidator(board, 3);
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+
+        //act
+        boolean isGameOver = boardValidator.saysItsOver();
+
+        //assert
+        assertTrue(isGameOver);
+    }
+
+    @Test(dataProvider = "winningHorizontallyWith4On4x4")
+    public void testHorizontalGameWinWhenConditionIs4On4x4Board(int[] chosenFields) throws NumberLowerThanOneException {
+        //arrange
+        Board board = new Board(4);
+        BoardValidator boardValidator = new BoardValidator(board, 4);
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[3]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[3]), Symbol.CROSS));
+
+
+        //act
+        boolean isGameOver = boardValidator.saysItsOver();
+
+        //assert
+        assertTrue(isGameOver);
+    }
+
+    @Test(dataProvider = "markRowOfThreeFieldsBeginningFromSecond")
+    public void testHorizontalGameWinWhenConditionIs3nTheMiddleOf5x5BoardRow(int[] chosenFields) throws NumberLowerThanOneException {
+        //arrange
+        Board board = new Board(5);
+        BoardValidator boardValidator = new BoardValidator(board, 3);
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[0]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[1]), Symbol.CROSS));
+        board.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+        boardValidator.handle(new Move(FieldNumber.valueOf(chosenFields[2]), Symbol.CROSS));
+
+        //act
+        boolean isGameOver = boardValidator.saysItsOver();
+
+        //assert
+        assertTrue(isGameOver);
+    }
```

Comment:
This tests are very difficult to understand. 

Also, they seem very similar so you could try to extract some common parts.

Reply:
They will be splitted into classes with DataProviders and Board / Move / FieldNumbers will be mocked as it should be.